src/main/java/com/aws/greengrass/logmanager/LogManagerService.java

```diff
@@ -504,68 +505,58 @@ private void handleCloudWatchAttemptStatus(CloudWatchAttempt cloudWatchAttempt)
-                // TODO: this is temporary for not changing the original logic, will be well handled.
-                if (!file1.isPresent()) {
-                    logger.atDebug().log("The fileHash does not exist in directory");
-                    return;
-                }
-                file = file1.get();
-                fileName = file.getAbsolutePath();
-                // TODO: the following logic is only for passing this small PR while not changing the current context.
-                //  We will grab the fileHash in the future.
-                Optional<LogFile> activeFile = logFileGroup.getActiveFile();
-                if (activeFile.isPresent()) {
-                    isActiveFile = activeFile.get().fileEquals(file);
-                }
-            } catch (InvalidLogGroupException e) {
-                logger.atDebug().cause(e).log();
+        try {
+            LogFileGroup logFileGroup = attemptLogInformation.getLogFileGroup().syncDirectory();
+            if (!logFileGroup.isHashExist(fileHash)) {
+                logger.atTrace().kv("fileHash", fileHash).log("component",
+                        logFileGroup.getFilePattern(), "File not found in directory");
                 return;
             }
-        }
-        // If we have completely read the file, then we need add it to the completed files list and remove it
-        // it (if necessary) for the current processing list.
-        String componentName = attemptLogInformation.getComponentName();
-        if (!isActiveFile && file.length() == cloudWatchAttemptLogFileInformation.getBytesRead()
-                + cloudWatchAttemptLogFileInformation.getStartPosition()) {
-            Set<String> completedFileNames = completedLogFilePerComponent.getOrDefault(componentName, new HashSet<>());
-            completedFileNames.add(fileName);
-            completedLogFilePerComponent.put(componentName, completedFileNames);
-            if (currentProcessingLogFilePerComponent.containsKey(componentName)) {
-                CurrentProcessingFileInformation fileInformation = currentProcessingLogFilePerComponent
-                        .get(componentName);
-                if (fileInformation.fileName.equals(fileName)) {
-                    currentProcessingLogFilePerComponent.remove(componentName);
+            LogFile file = logFileGroup.getFile(fileHash);
+            // @deprecated  This is deprecated value in versions greater than 2.2, but keep it here to avoid
+            // upgrade-downgrade issues.
+            String fileName = file.getAbsolutePath();
+            // If we have completely read the file, then we need add it to the completed files list and remove it
+            // it (if necessary) for the current processing list.
+            String componentName = attemptLogInformation.getComponentName();
+            if (!logFileGroup.isActiveFile(file) && file.length() == cloudWatchAttemptLogFileInformation.getBytesRead()
```

Comment:
nit: maybe change `==` to `<=`? This seems like a very delicate logic. If somehow bytes calculating went wrong then a file could be stuck in processing forever.

Reply:
Will add it in next PR.